packages/material-ui-system/src/styleFunctionSx/extendSxProp.js

```diff
@@ -0,0 +1,31 @@
+import { deepmerge } from '@material-ui/utils';
+import { propToStyleFunction } from '../getThemeValue';
+
+const splitProps = (props) => {
+  const result = {
+    systemProps: {},
+    otherProps: {},
+  };
+
+  Object.keys(props).forEach((prop) => {
+    if (propToStyleFunction[prop]) {
+      result.systemProps[prop] = props[prop];
+    } else {
+      result.otherProps[prop] = props[prop];
+    }
+  });
+
+  return result;
+};
+
+export default function extendSxProp(props) {
+  const { sx: inSx = {}, ...other } = props;
+  const { systemProps, otherProps } = splitProps(other);
+
+  const sx = systemProps ? deepmerge(systemProps, inSx) : inSx;
```

Comment:
Why a deep merge? It seems that systemProps will be flat or need to be treated as flat.

Reply:
`systemProps` seems always true-ish.

Reply:
The only reason for adding deepmerge was so that brekapoints can be merged, but I don't think we should even support it. If something is defined in `sx` it will replace the prop value and that should be it. Will change it to not use `deepmerge`.